### ISSUE TITLE
Excuse me, why do we still need to use expert grids for the testing dataset? Is it possible to implement grid partitioning for any step file?

### ISSUE THREAD

**Issue by user-deng** [NiklasFreymuth/AMBER#4]

**Comment by NiklasFreymuth** [comment]
The code requires expert meshes for all training/evaluation/testing data for evaluation purposes. If you want to use a trained AMBER model to generate meshes on geometries for which you do not have any expert data, its probably best to write a designated evaluation script that loads the model and does inference on a given input file.

**Comment by user-deng** [comment]
Thank you for your reply. What role can AMBER play in the process of generating grids for any step file?

**Comment by NiklasFreymuth** [comment]
After training, AMBER can be used to generate an adaptive meshes for unseen geometries through a series of intermediate meshes and sizing field predictions on the elements of these meshes. If you train it on some examples for your family of problems, it should generalize to new problems during inference

**Comment by user-deng** [comment]
Thank you very much for your reply. I will try my best to give it a try.

**Comment by user-deng** [comment]
![Snipaste_2024-08-02_09-59-39](https://github.com/user-attachments/assets/70851541-eb29-47c5-8583-bdd3b1860b2e)
Hello, I have encountered a new issue where when loading the model, I need to use 'exemplar_graph' and the data must be loaded in advance. For any step file, should I randomly select an "example_graph" or how can I solve it?

**Comment by user-deng** [comment]
![微信截图_20240802105855](https://github.com/user-attachments/assets/32abc2d1-9195-4165-8580-923b72c21479)
The SupervisedHMPN class requires the 'exemplar_graph' parameter, but your 'load_from_math' function says' kwargs' is optional. I don't know how to handle this, could you please give me some guidance?
![12](https://github.com/user-attachments/assets/8921ce0d-a553-43ce-82d1-d94e9ddbcbb2)

**Comment by user-deng** [comment]
def mesh_to_graph(wrapped_mesh: MeshWrapper, labels: np.ndarray, element_feature_names: List[str],
                  edge_feature_names: List[str], fem_problem: Optional[AbstractFiniteElementProblem],
                  device: str) -> Union[Data, HeteroData]:
    """
    Generates an observation graph from a finite element problem and a sizing field.
    The graph is used as input for the GNN-based supervised learning algorithm.
    Args:
        wrapped_mesh: CachedMeshWrapper containing the mesh data as required for, e.g., the mesh connectivity and
            element midpoints
        fem_problem: FEM problem containing problem-specific information, such as boundary conditions and material
            May be None.
        labels: Label for the supervised learning algorithm. Will usually be the sizing field.
        element_feature_names: Names of the features to use for the element nodes
        edge_feature_names: Names of the features to use for the edges between the elements in the observation graph
        device: The (accelerator) device to use for the graph data
    Returns:

    """
    element_features = get_mesh_element_features(wrapped_mesh,
                                                 fem_problem=fem_problem,
                                                 element_feature_names=element_feature_names)
    edge_attr, edge_index = _get_mesh_element2element_features(wrapped_mesh, edge_feature_names)

    graph_dict = {
        "x": element_features,
        "y": torch.tensor(labels, dtype=torch.float32),
        "edge_index": edge_index,
        "edge_attr": edge_attr,
    }
    graph = Data(**graph_dict)
    graph = graph.to(device)
    return graph

To input the model, graph data is necessary, but generating graph data requires expert mesh, so it is impossible to partition only step

**Comment by user-deng** [comment]
您好，我有一个问题想要咨询您，希望您有时间可以解答一下。
        # 1. Predict values, clip to generate sizing field
        graph = graph.to(self.device)
        predictions = self._predict(graph, train=False)
        sizing_field = self._clip_sizing_field(predictions)
您的模型预测的时候，要求输入的数据必须是graph，
        expert_mesh = self.expert_meshes[fem_idx]
        labels = interpolate_sizing_field(
            coarse_mesh=wrapped_mesh,
            fine_mesh=expert_mesh,
            sizing_field_query_scope=self.sizing_field_query_scope,
            interpolation_type=self.sizing_field_interpolation_type,
        )
        graph = mesh_to_graph(wrapped_mesh=wrapped_mesh,
                              labels=labels,
                              element_feature_names=self.element_feature_names,
                              edge_feature_names=self.edge_feature_names,
                              fem_problem=fem_problem,
                              device=device)
        return graph
然而，在得到graph的时候，必须要有从expert mesh产生的labels。所以，无法实现对仅仅有CAD文件的数据进行网格划分？有其他方法可以解决这个labels的问题吗？
我将labels= np.zeros(mesh.num_elements) 全部设置成0，生成的结果很不理想，希望您能给一个解决方案，打扰您了，谢谢您！

**Comment by user-deng** [comment]
![image](https://github.com/user-attachments/assets/736c1b4f-343b-4b01-84d1-d2588096d44b)
If other expert mesh are used as labels, the generation effect is good for the data in the dataset, but the effect is very poor for other step files don't exist in dataset.  Is this reasonable?

**Comment by NiklasFreymuth** [comment]
Yes, that would be the expected generalization performance. The approach is essentially supervised learning, and requires i.i.d. data during training to properly generalize to unseen objects during inference.
